packages/features/calendar-cache/api/cron.ts

```diff
@@ -54,15 +68,34 @@ const handleCalendarsToUnwatch = async () => {
   const calendarsWithEventTypeIdsGroupedTogether = getUniqueCalendarsByExternalId(calendarsToUnwatch);
   const result = await Promise.allSettled(
     Object.entries(calendarsWithEventTypeIdsGroupedTogether).map(
-      async ([externalId, { eventTypeIds, credentialId, id }]) => {
-        if (!credentialId) {
+      async ([externalId, { eventTypeIds, credentialId, userId, delegationCredentialId, id }]) => {
+        if (!credentialId && !delegationCredentialId) {
           // So we don't retry on next cron run
-          await SelectedCalendarRepository.updateById(id, { error: "Missing credentialId" });
-          console.log("no credentialId for SelecedCalendar: ", id);
+
+          // FIXME: There could actually be multiple calendars with the same externalId and thus we need to technically update error for all of them
+          await SelectedCalendarRepository.updateById(id, {
+            error: "Missing credentialId and delegationCredentialId",
+          });
+          log.error("no credentialId and delegationCredentialId for SelectedCalendar: ", id);
           return;
         }
-        const cc = await CalendarCache.initFromCredentialId(credentialId);
-        await cc.unwatchCalendar({ calendarId: externalId, eventTypeIds });
+
+        try {
+          const cc = await CalendarCache.initFromDelegationOrRegularCredential({
+            credentialId,
+            delegationCredentialId,
+            userId,
+          });
```

Comment:
Again we don't just have credentialId always, we could have delegationCredentialId as well which along with userId, identifies the exact credential